### ISSUE TITLE
method/@public as xs:boolean does not work when other degrees of visibility are desired

### ISSUE THREAD

**Issue by llorllale** [yegor256/jpeek#188]
**What I'm trying to do**
Implement `TCC`, which cares about methods of a class that are visible to other classes. This would include default and `protected` access levels, not just `public`

**What I'm getting**
The skeleton only provides `method/@public` indicating whether it is public or not. For TCC, I would like to know whether the method is private or not.

**My proposal**
In `skeleton` we substitute `method/@public` for `method/@visibility` with possible values: `private`, `default`, `protected`, and `public`.

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by yegor256** [comment]
@llorllale I didn't understand what you're suggesting to do...

**Comment by llorllale** [comment]
@yegor256 sorry, I updated the description

**Comment by yegor256** [comment]
@llorllale way better now, thanks!

**Comment by 0crat** [comment]
Job [#188](https://github.com/yegor256/jpeek/issues/188) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://datum.zerocracy.com/pages/policy.html#29): +15 points just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale), total is [+570](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
The job [#188](https://github.com/yegor256/jpeek/issues/188) assigned to @oridan[/z](https://www.0crat.com/u/oridan), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/1870722196); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
@oridan[/z](https://www.0crat.com/u/oridan) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by oridan** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/jpeek/issues/188#issuecomment-405062222))

@oridan The user @oridan[/z](https://www.0crat.com/u/oridan) resigned from [#188](https://github.com/yegor256/jpeek/issues/188), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @oridan[/z](https://www.0crat.com/u/oridan)

**Comment by 0crat** [comment]
The job [#188](https://github.com/yegor256/jpeek/issues/188) assigned to @butenkos[/z](https://www.0crat.com/u/butenkos), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/275d8e8d-3f97-4265-90e7-eb2e6a0319aa); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by butenkos** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/jpeek/issues/188#issuecomment-443644850))

@butenkos The user @butenkos[/z](https://www.0crat.com/u/butenkos) resigned from [#188](https://github.com/yegor256/jpeek/issues/188), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @butenkos[/z](https://www.0crat.com/u/butenkos)

**Comment by 0crat** [comment]
The job [#188](https://github.com/yegor256/jpeek/issues/188) assigned to @serranya[/z](https://www.0crat.com/u/serranya), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/e7ef0751-da85-4dd4-81f9-4f59615be63c); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by 0pdd** [comment]
@llorllale the puzzle [#297](https://github.com/yegor256/jpeek/issues/297) is still not solved.

**Comment by 0crat** [comment]
@rultor[/z](https://www.0crat.com/u/rultor) the issue is closed not by @llorllale[/z](https://www.0crat.com/u/llorllale) (its creator); I won't close the order; please, re-open it and ask @llorllale[/z](https://www.0crat.com/u/llorllale) to close it

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) please, pay attention to this issue

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/C7JGJ00DP) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by Serranya** [comment]
@0crat wait The pr is merged but the issue somehow still open.

**Comment by 0crat** [comment]
> @0crat wait The pr is merged but the issue somehow still open. ([here](https://github.com/yegor256/jpeek/issues/188#issuecomment-453572685))

@Serranya The impediment for [#188](https://github.com/yegor256/jpeek/issues/188) was registered successfully by @serranya[/z](https://www.0crat.com/u/serranya)

**Comment by Serranya** [comment]
@llorllale did #291 fix your problem? If so, please close the ticket.

**Comment by llorllale** [comment]
@Serranya looks good, thanks

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@c513f920-c67d-428a-9a40-d51b3a4fa5db, thread: AsyncProc-8
com.jcabi.xml.StrictXML[124] java.lang.IllegalArgumentException: 1 error(s) in XML document: -1:-1: Duplicate unique value [gh:yegor256/jpeek#188] declared for identity constraint of element "verbosity".
```

[0.40.4](https://github.com/zerocracy/farm): CID: [80d396a6-2e83-41a7-b5e9-a26bf08995ac](https://www.0crat.com/footprint/C7JGJ00DP/80d396a6-2e83-41a7-b5e9-a26bf08995ac), Type: "Close job"

**Comment by 0crat** [comment]
@cucumber007[/z](https://www.0crat.com/u/cucumber007) please review this job completed by @serranya[/z](https://www.0crat.com/u/serranya), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by cucumber007** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @cucumber007[/z](https://www.0crat.com/u/cucumber007)

**Comment by 0pdd** [comment]
@llorllale the puzzle [#359](https://github.com/yegor256/jpeek/issues/359) is still not solved; solved: [#297](https://github.com/yegor256/jpeek/issues/297).

**Comment by 0pdd** [comment]
@llorllale all 2 puzzles are solved here: [#297](https://github.com/yegor256/jpeek/issues/297), [#359](https://github.com/yegor256/jpeek/issues/359).
